compiler/passes/convert-uast.cpp

```diff
@@ -1414,6 +1415,12 @@ struct Converter {
     if (!node->isExpressionLevel()) return false;
     if (node->iterand()->isZip()) return false;
     if (node->numStmts() != 1) return false;
+
+    // This seems counterintuitive, as a bracket loop cannot possibly
+    // contain an index. However the old AST printer uses the
+    // 'maybeArrayType' flag in order to decide if something is a
+    // bracket loop or not. If we adjust that, we can probably remove
+    // this check. TODO: Maybe we should include a 'BracketLoop' type?
```

Comment:
These changes to convert-uast.cpp seem harmless but also unrelated. Might be better to have them in another PR.

Reply:
K